binary.go

```diff
@@ -16,19 +16,26 @@ type BitsWriter struct {
 	cacheLen byte
 	bsCache  []byte
 	w        io.Writer
+	writeCb  BitsWriterWriteCallback
 }
 
+type BitsWriterWriteCallback func([]byte)
+
 // BitsWriterOptions represents BitsWriter options
 type BitsWriterOptions struct {
 	ByteOrder binary.ByteOrder
 	Writer    io.Writer
+
+	// WriteCallback is called every time when
+	WriteCallback BitsWriterWriteCallback
```

Comment:
Could you move this attribute above `Writer` and make sure the comment is complete?

Reply:
whoa how come i haven't completed that comment :) done